### PR TITLE
fix: nested theme providers by removing menu's portal

### PR DESCRIPTION
**Pull request by dbanksdesign**
#### Description of changes

* Menu primitive uses `portalled={false}` now so it no longer renders in a React portal outside the DOM node of the ThemeProvider, meaning it now properly accepts the theme of the nearest ThemeProvider. 
* Removing all the junk in the ThemeProvider to get around that Portal issue. 

Potential alternative: we could make our own Portal (using the ThemeProvider's DOM node) potentially and render the menu content in that. Seems a bit more difficult. 

#### Description of how you validated changes

* Navigated around the docs site, things work fine. Examples with ThemeProvider work.

CleanShot 2022-06-14 at 11 10 15

CleanShot 2022-06-14 at 11 10 42

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by reesscot — APPROVED** [review]
This is great!

File: packages/react/src/components/ThemeProvider/index.tsx [hunk 1]
```diff
@@ -24,38 +24,6 @@ export function AmplifyProvider({
...
-          'data-amplify-theme',
-          originalName
-        );
-        document.documentElement.setAttribute(
-          'data-amplify-color-mode',
-          originalColorMode
-        );
-      };
-    }
-  }, [name, colorMode]);
```
Comment (calebpollman) [inline packages/react/src/components/ThemeProvider/index.tsx:58]: ✨

**Review by calebpollman — APPROVED** [review]
🚢

**Review by joebuono — APPROVED** [review]

File: packages/react/src/components/ThemeProvider/__tests__/ThemeProvider.test.tsx [hunk 2]
```diff
@@ -21,22 +21,16 @@ describe('ThemeProvider', () => {
...
   it('wraps the App in [data-amplify-theme="default-theme"]', () => {
-    jest.spyOn(global.document.documentElement, 'setAttribute');
     const { container } = render(
       <ThemeProvider>
         <App />
       </ThemeProvider>
     );
 
-    expect(global.document.documentElement.setAttribute).toHaveBeenCalledWith(
+    expect(container.querySelector(`[data-amplify-theme]`)).toHaveAttribute(
```
Comment (zchenwei) [inline packages/react/src/components/ThemeProvider/__tests__/ThemeProvider.test.tsx:30]:
```suggestion
    expect(global.document.documentElement).toHaveAttribute(
```
No need to query? The root element is kinda special

File: packages/react/src/components/ThemeProvider/index.tsx (hunk 1 above)
Comment (zchenwei) [reply packages/react/src/components/ThemeProvider/index.tsx:58]: 🤩

**Review by jacoblogan — APPROVED** [review]
Awesome this makes things so much simpler!!!

**Review by zchenwei — APPROVED** [review]
This is looking good. I am going to close #2081. Regarding the potential solution, radix has a Portal util that can help us achieve it easily with `containerRef` to customize the container - https://www.radix-ui.com/docs/primitives/utilities/portal#root

**Pull request merged**